spark3/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -42,7 +44,8 @@
   private static final ProcedureParameter[] PARAMETERS = new ProcedureParameter[] {
       ProcedureParameter.required("table", DataTypes.StringType),
       ProcedureParameter.optional("older_than", DataTypes.TimestampType),
-      ProcedureParameter.optional("retain_last", DataTypes.IntegerType)
+      ProcedureParameter.optional("retain_last", DataTypes.IntegerType),
+      ProcedureParameter.optional("num_parallelism", DataTypes.IntegerType)
```

Comment:
Nit (non-blocking): This parameter name is a bit confusing for me. Perhaps simply `num_threads` would do? It is indeed difficult to think of a name that does not evoke the normal spark task parallelism. Maybe even just `parallelism` would do?

Reply:
I agree with @kbendick that the name isn't very descriptive and parallelism can be mistaken with the number of shuffle partitions in Spark.

We could call it `delete_thread_pool_size` or `num_delete_threads`.

Thoughts, @karuppayya @kbendick @RussellSpitzer @raptond @flyrain @jackye1995?

Reply:
concurrent_deletes?

Reply:
I like that, it is shorter. We could prepend `max_concurrent_deletes` to match naming in the rewrite action.